src/scenic/core/vectors.py

```diff
@@ -4,427 +4,659 @@
-		defaultStepSize (float): Default step size for `followFrom`; default 5.
-			This is an upper bound: more steps will be taken as needed to ensure that no
-			single step is longer than this value, but if the distance to travel is small
-			then the steps may be smaller.
-	"""
-	def __init__(self, name, value, minSteps=4, defaultStepSize=5):
-		self.name = name
-		self.value = value
-		self.valueType = float
-		self.minSteps = minSteps
-		self.defaultStepSize = defaultStepSize
-
-	@distributionMethod
-	def __getitem__(self, pos) -> float:
-		return self.value(pos)
-
-	@vectorDistributionMethod
-	def followFrom(self, pos, dist, steps=None, stepSize=None):
-		"""Follow the field from a point for a given distance.
-
-		Uses the forward Euler approximation, covering the given distance with
-		equal-size steps. The number of steps can be given manually, or computed
-		automatically from a desired step size.
-
-		Arguments:
-			pos (`Vector`): point to start from.
-			dist (float): distance to travel.
-			steps (int): number of steps to take, or :obj:`None` to compute the number of
-				steps based on the distance (default :obj:`None`).
-			stepSize (float): length used to compute how many steps to take, or
-				:obj:`None` to use the field's default step size.
-		"""
-		if steps is None:
-			steps = self.minSteps
-			stepSize = self.defaultStepSize if stepSize is None else stepSize
-			if stepSize is not None:
-				steps = max(steps, math.ceil(dist / stepSize))
-
-		step = dist / steps
-		for i in range(steps):
-			pos = pos.offsetRadially(step, self[pos])
-		return pos
-
-	@staticmethod
-	def forUnionOf(regions):
-		"""Creates a `PiecewiseVectorField` from the union of the given regions.
-
-		If none of the regions have an orientation, returns :obj:`None` instead.
-		"""
-		if any(reg.orientation for reg in regions):
-			return PiecewiseVectorField('Union', regions)
-		else:
-			return None
-
-	def __str__(self):
-		return f'<{type(self).__name__} {self.name}>'
+    """A vector field, providing a heading at every point.
+
+    Arguments:
+        name (str): name for debugging.
+        value: function computing the heading at the given `Vector`.
+        minSteps (int): Minimum number of steps for `followFrom`; default 4.
+        defaultStepSize (float): Default step size for `followFrom`; default 5.
+            This is an upper bound: more steps will be taken as needed to ensure that no
+            single step is longer than this value, but if the distance to travel is small
+            then the steps may be smaller.
+    """
+    def __init__(self, name, value, minSteps=4, defaultStepSize=5):
+        self.name = name
+        self.value = value
+        self.valueType = Orientation
+        self.minSteps = minSteps
+        self.defaultStepSize = defaultStepSize
+
+    @distributionMethod
+    def __getitem__(self, pos) -> Orientation:
+        val = self.value(pos)
+        
+        if isinstance(val, (int, float)):
```

Comment:
Would be better to use `numbers.Real` here in case someone is using NumPy scalar types.